app/src/main/java/fr/neamar/kiss/result/Result.java

```diff
@@ -88,16 +73,28 @@ else if (pojo instanceof ShortcutsPojo)
     }
 
     /**
-     * Enrich text for display. Put text requiring highlighting between {}
+     * Enrich text for display.
+     *
      *
-     * @param text to highlight
-     * @return text displayable on a textview
+     * @param text Text to highlight
+     * @param positions List of matched positions
+     * @param context Application context
+     * @return Spannable displayable on a TextView
      */
-    static Spanned enrichText(String text, Context context) {
-        //TODO: cache the result. We consume lots of CPU and RAM converting every time we display
-        return Html.fromHtml(text.replaceAll("\\{", "<font color=" + UIColors.getPrimaryColor(context) + ">").replaceAll("\\}", "</font>"));
+    static Spanned enrichText(String text, List<Pair<Integer, Integer>> positions, Context context) {
+        SpannableString enriched = new SpannableString(text);
+
+        int primaryColor = UIColors.getPrimaryColor(context);
+        for (Pair<Integer, Integer> position : positions) {
+            enriched.setSpan(
+                    new ForegroundColorSpan(primaryColor),
```

Comment:
do you need to create a new one everytime?

Reply:
My idea here is that we could store this foregroundColorSpan as a static property, set it the first time and reuse it after.

When we change the color, from SettingsActivity, we can then set it back to null to ensure it gets recreated on color change. That way, we don't need to call getPrimaryColor() or create a new instance for every display

Reply:
Unfortunately no, you cannot reuse the same `ForegroundColorSpan` multiple times in the same `Spannable`. However, you may be able to re-use one across Spannable, so I don't know. I'll try something.